plugins/nf-validation/src/main/nextflow/validation/SamplesheetConverter.groovy

```diff
@@ -32,251 +32,93 @@ import nextflow.Session
-                    }
-                }
-                else {
-                    def inputVal = (input != '' && input != null) ?
-                            castToNFType(input, field) :
-                        field['value']['default'] != null ?
-                            castToNFType(field['value']['default'], field) :
-                            []
-                    output.add(inputVal)
-                }
-            }
-            // Add meta to the output when a meta field has been created
-            if(meta != [:]) { output.add(0, meta) }
-            return output
-        }
-
-        // check for samplesheet errors
-        if (this.hasErrors()) {
-            String message = "Samplesheet errors:\n" + this.getErrors().join("\n")
-            throw new SchemaValidationException(message, this.getErrors() as List)
-        }
-
-        // check for schema errors
-        if (this.hasSchemaErrors()) {
-            String message = "Samplesheet schema errors:\n" + this.getSchemaErrors().join("\n")
-            throw new SchemaValidationException(message, this.getSchemaErrors() as List)
-        }
-
-        // check for warnings
-        if( this.hasWarnings() ) {
-            def msg = "Samplesheet warnings:\n" + this.getWarnings().join('\n')
-            log.warn(msg)
-        }
-
-        return outputs
-    }
-
-    // Function to transform an input field from the samplesheet to its desired type
-    private static castToNFType(
-        Object input,
-        Map.Entry<String, Map> field
-    ) {
-        def String type = field['value']['type']
-        def String key = field.key
-
-        // Recursively call this function for each item in the array if the field is an array-type
-        // The returned values are collected into a single array
-        if (type == "array") {
-            def Map.Entry<String, Map> subfield = (Map.Entry<String, Map>) Map.entry(field.key, field['value']['items'])
-            log.debug "subfield = $subfield"
-            def ArrayList result = input.collect{ castToNFType(it, subfield) } as ArrayList
             return result
         }
+        return channelFormat
+    }
 
-        def String inputStr = input as String
-        // Convert string values
-        if(type == "string" || !type) {
-            def String result = inputStr as String
-            
-            // Check and convert to the desired format
-            def String format = field['value']['format']
-            if(format) {
-                if(format == "file-path-pattern") {
-                    def ArrayList inputFiles = Nextflow.file(inputStr) as ArrayList
-                    return inputFiles
-                }
-                if(format.contains("path")) {
-                    def Path inputFile = Nextflow.file(inputStr) as Path
-                    return inputFile
+    // TODO add path casting
+    private static Object formatEntry(Object input, LinkedHashMap schema) {
+
```

Comment:
Remove `TODO` and add some code comments to explain the logic of this method.

I think the nested/recursive nature of this method can make it easy to get lost.